query-service-impl/src/main/java/org/hypertrace/core/query/service/AdditionalHandlerFiltersConfig.java

```diff
@@ -0,0 +1,128 @@
+package org.hypertrace.core.query.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.util.JsonFormat;
+import com.typesafe.config.Config;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.experimental.NonFinal;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Operator;
+
+@Slf4j
+public class AdditionalHandlerFiltersConfig {
+  private static final String ADDITIONAL_TENANT_FILTERS_CONFIG_KEY = "additionalTenantFilters";
+  private final Map<String, List<Filter>> tenantToAdditionalFiltersMap;
+
+  public AdditionalHandlerFiltersConfig(Config config, Optional<String> timeFilterColumnName) {
+    if (config.hasPath(ADDITIONAL_TENANT_FILTERS_CONFIG_KEY)) {
+      this.tenantToAdditionalFiltersMap =
+          config.getConfigList(ADDITIONAL_TENANT_FILTERS_CONFIG_KEY).stream()
+              .map(filterConfig -> new TenantFilters(filterConfig, timeFilterColumnName))
+              .collect(
+                  Collectors.toMap(
+                      tenantFilters -> tenantFilters.tenantId,
+                      tenantFilters -> tenantFilters.filters));
+    } else {
+      this.tenantToAdditionalFiltersMap = Collections.emptyMap();
+    }
+  }
+
+  public List<Filter> getAdditionalFiltersForTenant(String tenantId) {
+    return this.tenantToAdditionalFiltersMap.getOrDefault(tenantId, Collections.emptyList());
+  }
+
+  @lombok.Value
+  @NonFinal
+  private static class TenantFilters {
+    private static final String TENANT_ID_CONFIG_KEY = "tenantId";
+    private static final String TIME_RANGE_AND_FILTERS_CONFIG_KEY = "timeRangeAndFilters";
+    String tenantId;
+    List<Filter> filters;
+
+    private TenantFilters(Config config, Optional<String> startTimeAttributeName) {
+      this.tenantId = config.getString(TENANT_ID_CONFIG_KEY);
+      this.filters =
+          config.getConfigList(TIME_RANGE_AND_FILTERS_CONFIG_KEY).stream()
+              .map(TimeRangeAndFilter::new)
+              .map(
+                  timeRangeAndFilter ->
+                      timeRangeAndFilter.buildTimeRangeAndFilters(startTimeAttributeName))
+              .filter(Optional::isPresent)
+              .map(Optional::get)
+              .collect(Collectors.toList());
+    }
+  }
+
+  @lombok.Value
+  @NonFinal
+  private static class TimeRangeAndFilter {
```

Comment:
```suggestion
  private static class FiltersForTimeRange {
```
The existing name `TimeRangeAndFilter ` is quite confusing. It sounds like it is carrying 2 unrelated concepts together: timeRange and filters.